saleor/dashboard/category/views.py

```diff
@@ -110,7 +112,10 @@ def category_delete(request, pk):
     if request.method == "POST":
         descendants = category.get_descendants()
         menus = get_menus_that_need_update(categories=descendants)
+        products = unpublished_products_before_category_delete(category)
+        product_ids = [product.pk for product in products]
```

Comment:
```suggestion
        product_ids = list(products.values_list("id", flat=True))
```

Otherwise it is very heavy.